bpf/tests/common.h

```diff
@@ -272,4 +272,27 @@ _Pragma("unroll")						\
 	return 0;						\
 }
 
+static void *(*map_lookup_percpu_elem)(void *map, const void *key, unsigned int cpu) = (void *)195;
+
+/* Asserts that the per-cpu metrics map contains at least one entry that */
+/* contains a specified count given a metrics_key.*/
+#define assert_metrics_contains(key, count) \
+({ \
+	struct metrics_value *__entry = NULL; \
+	bool found = false; \
+	/* Iterate until lookup encounters null when hitting cpu number */ \
+	/* Assumes at most 128 CPUS */ \
+	for (int i = 0; i < 128; i++) { \
+		__entry = map_lookup_percpu_elem(&METRICS_MAP, &key, i); \
+		if (!__entry) { \
+			break; \
+		} \
+		if (__entry->count == count) { \
```

Comment:
Maybe we should accumulate `__entry->count` and compare the `count` with summed up `__entry->count`?

Something like:
```suggestion
__u64 sum;
for (int i = 0; i < 128; i++) {
    __entry = map_lookup_percpu_elem(&METRICS_MAP, &key, i);
    if (!__entry) break;
    sum += __entry->count;
}
if (sum == count) found=true;
```

Reply:
I think it depends on the semantics of macro. The macro we create here is `assert_metrics_contains`, so your implementation above is accurate. My suggested code is more like `assert_metrics_count`.

Reply:
This makes sense, made the change 👍